OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -236,20 +236,18 @@ public override bool HandleKeyPress(KeyInput e)
 				}
 				else if (key == Game.Settings.Keys.SelectUnitsByTypeKey)
 				{
-					var selectedTypes = World.Selection.Actors
+					var selectedClasses = World.Selection.Actors
 						.Where(x => x.Owner == player)
-						.Select(a => a.Info);
+						.Select(a => a.Trait<Selectable>().Class);
 
-					Func<Actor, bool> cond = a => a.Owner == player && selectedTypes.Contains(a.Info);
-					var tl = worldRenderer.Viewport.TopLeft;
-					var br = worldRenderer.Viewport.BottomRight;
-					var newSelection = SelectActorsInBox(World, tl, br, cond);
+					Func<Actor, bool> cond = a => a.Owner == player && selectedClasses.Contains(a.Trait<Selectable>().Class);
+					var newSelection = SelectActorsInBox(World, worldRenderer.Viewport.TopLeft, worldRenderer.Viewport.BottomRight, cond);
 
-					if (newSelection.Count() > selectedTypes.Count())
+					if (newSelection.Count() > selectedClasses.Count())
```

Comment:
This makes no sense, and will likely break when you add `.ToHashSet()` above.
I think this should be `newSelection.Count() > World.Selection.Actors.Count()`. Can someone confirm?